tests/apollo/util/skvbc.py

```diff
@@ -49,10 +49,35 @@ class SimpleKVBCProtocol:
     requests.
     """
 
-    def __init__(self, bft_network, tracker = None, pre_process=False):
+    def __init__(self, bft_network, tracker = None, pre_exec_all = None):
```

Comment:
Hi Alex, to simplify things I suggest you removed the optional `pre_exec_all` parameter altogether. 
The value should always come from the `tracker` object.

Reply:
I did consider removing this parameter entirely when developing this change, though I was not sure whether or not I should do it. `tracker` is also an optional parameter, and `SimpleKVBCProtocol` currently [does use the value](https://github.com/vmware/concord-bft/blob/master/tests/apollo/util/skvbc.py#L416) of its copy of `pre_exec_all`, regardless of whether or not`tracker` is `None`. We could still get rid of the `pre_exec_all` parameter here and just set the `SimpleKVBCProtocol`'s `pre_exec_all` attribute to `False` if no `tracker` is given, and, as far as I can tell, that would not break or otherwise affect any of the existing test cases, as I am not aware of any instances where a `SimpleKVBCProtocol` is instantiated with no `tracker` and `True` for `pre_exec_all`. However, I had some slight reservation about doing this, as it looks like `SimpleKVBCProtocol` currently does theoretically support testing with pre-execution and no history tracker, and I don't know whether we might add any test cases that actually do this in the future.

Do you have any sense of whether we intend for `SimpleKVBCProtocol` to be supporting initialization with `tracker=None` and `pre_exec_all=True` and/or what we anticipate the possibility is of future test cases that would want to test with pre-execution and no history tracker?

Reply:
Hi Alex, thanks for pointing this out. You are right,  we do want to preserve the option to test with pre-execution, but without a tracker.

So  I am withdrawing my concern and approve your changes for merging as they are.